api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -12,41 +12,108 @@
 import org.openmrs.Location;
 import org.openmrs.Patient;
 import org.openmrs.Provider;
-import org.openmrs.api.APIException;
 import org.openmrs.api.impl.BaseOpenmrsService;
-import org.openmrs.module.patientqueueing.PatientQueueingConfig;
+import org.openmrs.module.patientqueueing.QueueingUtils;
 import org.openmrs.module.patientqueueing.api.PatientQueueingService;
 import org.openmrs.module.patientqueueing.api.dao.PatientQueueingDao;
 import org.openmrs.module.patientqueueing.model.PatientQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.text.ParseException;
+import java.util.ArrayList;
 import java.util.Date;
 import java.util.List;
 
 public class PatientQueueingServiceImpl extends BaseOpenmrsService implements PatientQueueingService {
 	
+	private static final Logger log = LoggerFactory.getLogger(PatientQueueingServiceImpl.class);
+	
 	PatientQueueingDao dao;
 	
 	public void setDao(PatientQueueingDao dao) {
 		this.dao = dao;
 	}
 	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getPatientQueueByQueueNumber(java.lang.String)
+	 */
+	@Override
+	public PatientQueue getPatientQueueByQueueNumber(String queueNumber) {
+		List<PatientQueue> patientQueueList = dao.getPatientQueueByQueueNumber(queueNumber);
+		if (!patientQueueList.isEmpty()) {
+			return patientQueueList.get(0);
+		} else {
+			return null;
+		}
+		
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#generateQueueNumber(org.openmrs.Location,
+	 *      java.util.Date, org.openmrs.Patient)
+	 */
+	public String generateQueueNumber(Location location, Date date, Patient patient) {
```

Comment:
What plan do you have for those who would like a different format of queue numbers?

Reply:
Its complicated to write a generator generic enough for all possible scenarios of queueNumber format. Incase of a special case they should be creating their own generator .

Reply:
How do their new generator plug into your framework?